### PR TITLE
Add hyperstart-exec sync wrapper

### PR DESCRIPTION
**Pull request by Crazykev**
Move first commit from #409, just make this method public.
PTAL @laijs @gao-feng 

Signed-off-by: Crazykev <crazykev@zju.edu.cn>

### CODE REVIEW

File: hypervisor/vm.go [hunk 1]
```diff
@@ -481,6 +483,47 @@ func (vm *Vm) OnlineCpuMem() error {
...
+	if err != nil {
+		return nil, nil, err
+	}
+
+	result := vm.WaitProcess(false, []string{execId}, -1)
+	if result == nil {
+		err = fmt.Errorf("can not wait hyperstart-exec %q", execId)
+		glog.Error(err)
+		return nil, nil, err
+	}
```
Comment (laijs) [inline hypervisor/vm.go]: wrap WaitProcess() in a go routine and move it up

**Comment by Crazykev** [comment]
@laijs Updated, pls check.

File: hypervisor/vm.go [hunk 2]
```diff
@@ -481,6 +483,60 @@ func (vm *Vm) OnlineCpuMem() error {
...
+
+	execId := fmt.Sprintf("hyperstart-exec-%s", utils.RandStr(10, "alpha"))
+
+	var stdoutBuf, stderrBuf bytes.Buffer
+	tty := &TtyIO{
+		Stdout: &stdoutBuf,
+		Stderr: &stderrBuf,
+	}
+	tty.Stdin = ioutil.NopCloser(bytes.NewReader(stdin))
```
Comment (laijs) [inline hypervisor/vm.go]: tty := &TtyIO{
                Stdin:   ioutil.NopCloser(bytes.NewReader(stdin)),
 	        Stdout: &stdoutBuf,
 		Stderr: &stderrBuf,
 	}

File: hypervisor/vm.go [hunk 3]
```diff
@@ -481,6 +483,60 @@ func (vm *Vm) OnlineCpuMem() error {
...
+
+	var stdoutBuf, stderrBuf bytes.Buffer
+	tty := &TtyIO{
+		Stdin:  ioutil.NopCloser(bytes.NewReader(stdin)),
+		Stdout: &stdoutBuf,
+		Stderr: &stderrBuf,
+	}
+
+	retErr := make(chan error, 1)
+	go func() {
```
Comment (laijs) [inline hypervisor/vm.go]: sorry, I was wrong, we don't need 'go func()' for WaitProcess()

File: hypervisor/vm.go [hunk 4]
```diff
@@ -481,6 +483,60 @@ func (vm *Vm) OnlineCpuMem() error {
...
+		}
+		close(retErr)
+	}()
+
+	err = vm.AddProcess(hyperstartapi.HYPERSTART_EXEC_CONTAINER, execId, false, cmd, []string{}, "/", tty)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	err, ok := <-retErr
```
Comment (laijs) [inline hypervisor/vm.go]: directly use `r, ok := <-result`

File: hypervisor/vm.go (hunk 3 above)
Comment (Crazykev) [inline hypervisor/vm.go]: I got your point, to avoid process finish before wait, should be worked now.

**Pull request merged**